### PR TITLE
[ranking] Adding initial path counts

### PR DESCRIPTION
**Pull request by Numbers88s**
Adds initial path counts so we can have the paths available even if there are no references.

## Test plan
Added test and tested manually.

### CODE REVIEW

File: enterprise/internal/codeintel/ranking/internal/store/ranking.go [hunk 1]
```diff
@@ -289,6 +289,55 @@ SELECT
...
+	if err != nil {
+		return err
+	}
+	defer func() { err = tx.Done(err) }()
+
+	inserter := func(inserter *batch.Inserter) error {
+		for _, path := range documentPath {
+			if err := inserter.Insert(ctx, repositoryID, path, 0, derivativeGraphKey); err != nil {
+				return err
+			}
```
Comment (efritz) [inline enterprise/internal/codeintel/ranking/internal/store/ranking.go:312]: repositoryID/0/derivativeGraphKey are all the same for ever iteration of the loop. If you create a temp table that has only a path field (then do an insert from temp table -> target table with the three other values) you can fit 4x more rows in a batch.

Comment (efritz) [reply enterprise/internal/codeintel/ranking/internal/store/ranking.go:312]: (Note this doesn't change the observable behavior of this function so your test should remain unchanged).

Comment (Numbers88s) [reply enterprise/internal/codeintel/ranking/internal/store/ranking.go:312]: Amazing! Will do.

File: enterprise/internal/codeintel/ranking/internal/store/ranking_test.go [hunk 2]
```diff
@@ -169,6 +169,38 @@ func TestInsertPathRanks(t *testing.T) {
 	}
 }
 
+func TestInsertInitialPathRanks(t *testing.T) {
```
Comment (efritz) [inline enterprise/internal/codeintel/ranking/internal/store/ranking_test.go:172]: - [ ] Also test `InsertInitialPathCounts`

File: migrations/frontend/1678899992_add_codeintel_ranking_paths/up.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+);
+
+CREATE TABLE IF NOT EXISTS codeintel_initial_path_ranks_processed(
+    id BIGSERIAL PRIMARY KEY,
+    graph_key TEXT NOT NULL,
+    codeintel_initial_path_ranks_id BIGINT NOT NULL,
+
+    CONSTRAINT fk_codeintel_initial_path_ranks FOREIGN KEY (codeintel_initial_path_ranks_id) REFERENCES codeintel_initial_path_ranks(id) ON DELETE CASCADE
+);
+
```
Comment (efritz) [inline migrations/frontend/1678899992_add_codeintel_ranking_paths/up.sql:15]:
```suggestion
```

File: enterprise/internal/codeintel/ranking/internal/store/ranking_test.go (hunk 2 above)
Comment (Numbers88s) [reply enterprise/internal/codeintel/ranking/internal/store/ranking_test.go:172]: Done ✅

**Comment by efritz** [comment]
@Numbers88s Let's merge in #49461 when you get a chance to review.

**Review by efritz — APPROVED** [review]

**Pull request merged**